v03_pipeline/lib/reference_data/dataset_table_operations.py

```diff
@@ -173,6 +177,113 @@ def annotate_dataset_globals(joined_ht: hl.Table, dataset: str, dataset_ht: hl.T
+) -> bool:
+    config = CONFIG[dataset][reference_genome.v02_value]
+    checks = {
+        'version': _ht_version_matches_config(
+            joined_ht,
+            dataset,
+            config,
+            reference_genome,
+        ),
+        'path': _ht_path_matches_config(joined_ht, dataset, config),
+        'enum': _ht_enums_match_config(joined_ht, dataset, config),
+        'select': _ht_selects_match_config(joined_ht, dataset, config),
+    }
+
+    results = []
+    for check, result in checks.items():
+        if result is False:
+            logger.info(f'{check} mismatch for {dataset}')
+
+        results.append(result)
+
+    return all(results)
+
+
+def _ht_version_matches_config(
+    joined_ht: hl.Table,
+    dataset: str,
+    dataset_config: dict,
+    reference_genome: ReferenceGenome,
+) -> bool:
+    joined_ht_version = hl.eval(
+        joined_ht.index_globals().versions.get(dataset, hl.missing(hl.tstr)),
+    )
+    config_version = dataset_config.get('version')
+
+    if joined_ht_version is None:
+        return False
+
+    if joined_ht_version and config_version:
+        return joined_ht_version == config_version
+
+    # if joined_ht_version and not config_version, compare joined ht version to annotated version in dataset ht
+    dataset_ht = import_ht_from_config_path(dataset_config, reference_genome)
+    annotated_version = hl.eval(
+        dataset_ht.globals.get('version', hl.missing(hl.tstr)),
+    )
+    return joined_ht_version == annotated_version
+
+
+def _ht_path_matches_config(
+    joined_ht: hl.Table,
+    dataset: str,
+    dataset_config: dict,
+) -> bool:
+    joined_ht_path = hl.eval(
+        joined_ht.index_globals().paths.get(dataset, hl.missing(hl.tstr)),
+    )
+
+    if joined_ht_path is None:
+        return False
+
+    config_path = (
+        dataset_config['source_path']
+        if 'custom_import' in dataset_config
+        else dataset_config['path']
+    )
+    return joined_ht_path == config_path
+
+
+def _ht_enums_match_config(
+    joined_ht: hl.Table,
+    dataset: str,
+    dataset_config: dict,
+) -> bool:
+    joined_ht_enums = hl.eval(
+        joined_ht.index_globals().enums.get(dataset, hl.Struct()),
+    )
+    config_enums = hl.eval(hl.struct(**dataset_config.get('enum_select', {})))
+
+    return hl.eval(joined_ht_enums == config_enums)
+
+
+def _ht_selects_match_config(
+    joined_ht: hl.Table,
+    dataset: str,
+    dataset_config: dict,
+) -> bool:
+    joined_ht_selects = set(joined_ht[dataset])
+
+    raw_config_selects = dataset_config.get('select', {})
+    config_selects = (
+        set(raw_config_selects)
+        if isinstance(raw_config_selects, list)
+        else set(raw_config_selects.keys())
+    )
+
+    if 'custom_select' in dataset_config:
+        config_selects.update(set(dataset_config.get('custom_select_keys')))
+
+    # check if all fields in 'config_selects' are in 'joined_ht_selects'
+    return config_selects.issubset(joined_ht_selects)
```

Comment:
Is the idea here that we would want to leave the joined_ht alone if we've removed a select?

Reply:
now that you bring this up, we should probably remove it from the joined_ht too, right?

Reply:
yep!  I think any change to the selects or custom selects should trigger a drop and re-join of the dataset from the `joined_ht`.

Reply:
instead of `issubset()`, I'm using `symmetric_difference()` which should trigger a drop and rejoin if either set contains a field not in the other